ou_dedetai/control.py

```diff
@@ -8,222 +8,167 @@
-    msg.status("Calculating backup size…", app=app)
-    t = utils.start_thread(utils.get_folder_group_size, src_dirs, q)
+    q: queue.Queue[int] = queue.Queue()
+    message = "Calculating backup size…"
+    app.status(message)
+    i = 0
+    t = app.start_thread(utils.get_folder_group_size, src_dirs, q)
     try:
         while t.is_alive():
-            msg.logos_progress()
+            i += 1
+            i = i % 20
+            app.status(f"{message}{"." * i}\r")
             time.sleep(0.5)
-        print()
     except KeyboardInterrupt:
         print()
-        msg.logos_error("Cancelled with Ctrl+C.", app=app)
+        app.exit("Cancelled with Ctrl+C.")
     t.join()
-    if config.DIALOG == 'tk':
-        app.root.event_generate('<<StopIndeterminateProgress>>')
-        app.root.event_generate('<<ClearStatus>>')
     src_size = q.get()
     if src_size == 0:
-        msg.logos_warning(f"Nothing to {mode}!", app=app)
-        return
+        app.exit(f"Nothing to {mode}!")
 
     # Set destination folder.
     if mode == 'restore':
-        dst_dir = Path(config.LOGOS_EXE).parent
+        if not app.conf.logos_exe:
+            app.exit("Cannot restore, Logos is not installed")
+        dst_dir = Path(app.conf.logos_exe).parent
         # Remove existing data.
         for d in data_dirs:
             dst = Path(dst_dir) / d
             if dst.is_dir():
                 shutil.rmtree(dst)
     else:  # backup mode
-        timestamp = config.get_timestamp().replace('-', '')
-        current_backup_name = f"{config.FLPRODUCT}{config.TARGETVERSION}-{timestamp}"  # noqa: E501
+        timestamp = utils.get_timestamp().replace('-', '')
+        current_backup_name = f"{app.conf.faithlife_product}{app.conf.faithlife_product_version}-{timestamp}"  # noqa: E501
         dst_dir = backup_dir / current_backup_name
         logging.debug(f"Backup directory path: \"{dst_dir}\".")
 
         # Check for existing backup.
         try:
             dst_dir.mkdir()
         except FileExistsError:
-            msg.logos_error(f"Backup already exists: {dst_dir}.")
+            # This shouldn't happen, there is a timestamp in the backup_dir name
+            app.exit(f"Backup already exists: {dst_dir}.")
 
     # Verify disk space.
     if not utils.enough_disk_space(dst_dir, src_size):
         dst_dir.rmdir()
-        msg.logos_warning(f"Not enough free disk space for {mode}.", app=app)
-        return
+        app.exit(f"Not enough free disk space for {mode}.")
 
     # Run file transfer.
     if mode == 'restore':
         m = f"Restoring backup from {str(source_dir_base)}…"
     else:
         m = f"Backing up to {str(dst_dir)}…"
-    msg.status(m, app=app)
-    msg.status("Calculating destination directory size", app=app)
-    dst_dir_size = utils.get_path_size(dst_dir)
-    msg.status("Starting backup…", app=app)
-    t = utils.start_thread(copy_data, src_dirs, dst_dir)
+    app.status(m)
+    app.status("Starting backup…")
```

Comment:
The 2nd status message should be `app.status(f"Starting {mode}…")`

Reply:
Oh, I see now that this is just a copy over from the previous code. Still seems redundant, though.

Reply:
Fixed in https://github.com/FaithLife-Community/LogosLinuxInstaller/pull/206/commits/3687c778e078ed4a8534594fc06f9f43c0700f57